src/utils/exists.js

```diff
@@ -1,22 +1,21 @@
-const fs = require("fs");
 /**
  * Check if file exists and cache the result
  * return the result in cache
  *
  * @example
- * var exists = require('./helpers/fsExists')({});
- * exists('.babelrc'); // false
+ * var exists = require(require('fs'), './helpers/fsExists')({});
+ * exists(require('fs'), '.babelrc'); // false
  */
-module.exports = function(cache) {
-  cache = cache || {};
+module.exports = function(fileSystem, filename) {
+  if (!filename) return false;
 
-  return function(filename) {
-    if (!filename) return false;
+  let exists = false;
 
-    cache[filename] =
-      cache[filename] ||
-      (fs.existsSync(filename) && fs.statSync(filename).isFile());
+  try {
+    exists = fileSystem.statSync(filename).isFile();
```

Comment:
Can we do `fs.existsSync(filename)` here instead of exception catching. Try catch will make v8 deoptimize this function.